src/sage/coding/linear_code.py

```diff
@@ -2382,13 +2399,62 @@ def _latex_(self):
         """
         return "[%s, %s]\\textnormal{ Linear code over }%s"\
             % (self.length(), self.dimension(), self.base_ring()._latex_())
+    
+    def intersection(self, other):
+        """
+        Return the intersection of this linear code with another.
+        The intersection of two linear codes C1 and C2 is the set of all
+        codewords that are in both C1 and C2. It is also a linear code.
+
+        This is computed using the identity:
+        C1 ∩ C2 = (C1_dual + C2_dual)_dual
+
+        INPUT:
+        - ``other`` -- a linear code.
+
+        OUTPUT:
+        - a linear code, the intersection of self and other.
 
+        EXAMPLES:
+            sage: F = GF(2)
+            sage: G1 = matrix(F, [[1,1,0,0], [0,0,1,1]])
+            sage: C1 = LinearCode(G1)
+            sage: G2 = matrix(F, [[1,0,1,0], [0,1,0,1]])
+            sage: C2 = LinearCode(G2)
+            sage: C_int = C1.intersection(C2)
+            sage: C_int
+            [4, 1] linear code over GF(2)
+            sage: c = vector(F, (1,1,1,1))
+            sage: c in C_int
+            True
+
+        # Test intersection with the zero code
```

Comment:
Should be `Test intersection with the zero code::` with a blank line after it. (The :: starts a code block in RST and without it, the doctest runner won't recognize this as a test.)